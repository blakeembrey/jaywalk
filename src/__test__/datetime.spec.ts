import { validationFunction } from '../'

describe('DateTime', () => {
  it('should validate date times', () => {
    const validate = validationFunction({
      '@type': 'DateTime'
    })

    const input = new Date()

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject non-date times', () => {
    const validate = validationFunction({
      '@type': 'DateTime'
    })

    expect.assertions(1)

    return validate(true).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })

  it('should be possible to specify min date', () => {
    const validate = validationFunction({
      '@type': 'DateTime',
      min: '2020-01-01'
    })

    return validate(new Date('2016-01-01')).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })

  it('should be possible to specify max date', () => {
    const validate = validationFunction({
      '@type': 'DateTime',
      max: '2020-01-01'
    })

    return validate(new Date('2025-01-01')).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })
})
