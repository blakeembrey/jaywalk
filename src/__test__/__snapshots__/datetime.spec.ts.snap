exports[`DateTime should be possible to specify max date 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "max date",
    "schemaPath": Array [
      "max",
    ],
  },
]
`;

exports[`DateTime should be possible to specify min date 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "min date",
    "schemaPath": Array [
      "min",
    ],
  },
]
`;

exports[`DateTime should reject non-date times 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "not a date",
    "schemaPath": Array [
      "@type",
    ],
  },
]
`;
